sql/core/src/test/scala/org/apache/spark/sql/execution/command/PlanResolutionSuite.scala

```diff
@@ -65,6 +65,26 @@ class PlanResolutionSuite extends AnalysisTest {
     }
   }
 
+  test("create table - partitioned by transforms") {
+    val transforms = Seq(
+        "bucket(16, b)", "years(ts)", "months(ts)", "days(ts)", "hours(ts)", "foo(a, 'bar', 34)",
+        "bucket(32, b), days(ts)")
```

Comment:
this reminds me one thing: shall we resolve the references before passing the transforms to the table catalog? Now we create transforms at parser side, and we may pass `years(ts)` to the table catalog even if `ts` doesn't exist. Do we expect the table catalog itself to resolve references?

Reply:
SGTM